docs/contributing/code-reviewing.md

```diff
@@ -1,231 +1,364 @@
-to update the GitHub thread to reflect that. Best practices are to:
-
-- Make sure that CI passes and the PR is rebased onto recent `main`.
-- Post comments on each feedback thread explaining at least how you
-  resolved the feedback, as well as any other useful information
-  (problems encountered, reasoning for why you picked one of several
-  options, a test you added to make sure the bug won't recur, etc.).
-- Mark any resolved threads as "resolved" in the GitHub UI, if
-  appropriate.
-- Post a summary comment in the main feed for the PR, explaining that
-  this is ready for another review, and summarizing any changes from
-  the previous version, details on how you tested the changes, new
-  screenshots/etc. More detail is better than less, as long as you
-  take the time to write clearly.
-
-If you resolve the feedback, but the PR has merge conflicts, CI
-failures, or the most recent comment is the reviewer asking you to fix
-something, it's very likely that a potential reviewer skimming your PR
-will assume it isn't ready for review and move on to other work.
-
-If you need help or think an open discussion topic requires more
-feedback or a more complex discussion, move the discussion to a topic
-in the Zulip development community server. Be sure to provide links
-from the GitHub PR to the conversation (and vice versa) so that it's
-convenient to read both conversations together.
+Code review is a key part of how Zulip does development. It's an essential aspect
+of our process to build a high-quality product with a maintainable code base.
 
 ## Principles of code review
 
-### Anyone can review
-
-Anyone can do a code review -- you don't have to have a ton of
-experience, and you don't have to have the power to ultimately merge
-the PR. If you
-
-- read the code, see if you understand what the change is
-  doing and why, and ask questions if you don't; or
+Zulip has an active contributor community, and just a small handful
+of maintainers who can do the final rounds of code review. As such, we would
+love for contributors to help each other with making pull requests that are not
+only correct, but easy to review. Doing so ensures that PRs can be finalized and
+merged more quickly, and accelerates the pace of progress for the entire
+project.
 
-- fetch the code (for Zulip server code,
-  [tools/fetch-rebase-pull-request][git tool] is super handy), play around
-  with it in your dev environment, and say what you think about how
-  the feature works
+### Reviewing your own code
 
-those are really helpful contributions.
+To start with, we strongly encourage you to get into the habit of reviewing you
+own code. You can often find things you missed by taking a step back to look
+over your work before asking others to do so, and this guide will walk you
+through the process. Catching mistakes yourself will help your PRs be merged
+faster, and folks will appreciate the quality and professionalism of your
+work.
```

Comment:
I think this needs a stronger motivating statement. I can take a pass at it later, but I think something more like "One of the best ways to improve the quality of your own work as a software engineer is to a code review of your own work before submitting it to others for review." is an important angle here.

Reply:
I took a pass at it, but you should take another look.